expression/builtin_time_vec.go

```diff
@@ -1721,11 +1722,52 @@ func (b *builtinAddDateDurationRealSig) vecEvalDuration(input *chunk.Chunk, resu
 }
 
 func (b *builtinSecToTimeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinSecToTimeSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalReal(b.ctx, input, buf); err != nil {
+		return err
+	}
+	args := buf.Float64s()
+	result.ResizeGoDuration(n, false)
+	result.MergeNulls(buf)
+	durations := result.GoDurations()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		secondsFloat := args[i]
+		negative := ""
+		if secondsFloat < 0 {
+			negative = "-"
+			secondsFloat = math.Abs(secondsFloat)
+		}
+		seconds := int64(secondsFloat)
+		demical := secondsFloat - float64(seconds)
+		hour := seconds / 3600
+		minute := seconds % 3600 / 60
+		second := seconds % 60
+		if hour > 838 {
+			hour = 838
+			minute = 59
+			second = 59
+		}
+		secondDemical := float64(second) + demical
+		duration, err := types.ParseDuration(b.ctx.GetSessionVars().StmtCtx, fmt.Sprintf("%s%02d:%02d:%v", negative, hour, minute, secondDemical), int8(b.tp.Decimal))
+		if err != nil {
+			result.SetNull(i, true)
+			return err
```

Comment:
We could delete the line 1765 and ```return err``` directly.

Reply:
Fixed.